### PR TITLE
support nested register derive & cleanups

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
r? Emilgardis

Partially fixes #610

File: src/generate/peripheral.rs [hunk 1]
```diff
@@ -262,6 +226,42 @@ pub fn render(
...
+        match (erc, ancestor.as_ref()) {
+            (Register(reg), Register(other_reg)) => {
+                Cow::Owned(Register(reg.derive_from(other_reg)))
+            }
+            (Cluster(cluster), Cluster(other_cluster)) => {
+                Cow::Owned(Cluster(cluster.derive_from(other_cluster)))
+            }
+            _ => {
+                return Err(anyhow!(
+                    "{} can't derive from {}",
```
Comment (Emilgardis) [inline src/generate/peripheral.rs:254]:
```suggestion
                    "{} can't be derived from {}",
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- Inside peripheral now supported multi-level `deriveFrom` for registers
```
Comment (Emilgardis) [inline CHANGELOG.md:10]:
```suggestion
- Support "nested" `deriveFrom` for registers
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
+- Use `expand_properties` from `svd-parser`
```
Comment (Emilgardis) [inline CHANGELOG.md:11]: don't see why this has to be added to the changelog

**Review by Emilgardis — CHANGES_REQUESTED** [review]

**Comment by burrbull** [comment]
изображение

MB9AF10xN.svd

Now it is correct

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**